kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/jms/DroolsJmsResponseHandlerIntegrationTest.java

```diff
@@ -0,0 +1,122 @@
+import org.kie.server.integrationtests.drools.DroolsKieServerBaseIntegrationTest;
+import org.kie.server.integrationtests.shared.KieServerAssert;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.kie.server.integrationtests.shared.KieServerAssert.*;
+
+@Category({JMSOnly.class})
+public class DroolsJmsResponseHandlerIntegrationTest extends DroolsKieServerBaseIntegrationTest {
+
+    private static ReleaseId releaseId = new ReleaseId("org.kie.server.testing", "ruleflow-group", "1.0.0.Final");
+
+    private static final String CONTAINER_ID = "ruleflow";
+    private static final String KIE_SESSION = "defaultKieSession";
+    private static final String PROCESS_ID = "simple-ruleflow";
+    private static final String LIST_NAME = "list";
+    private static final String LIST_OUTPUT_NAME = "output-list";
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static Collection<Object[]> data() {
+        KieServicesConfiguration jmsConfiguration = createKieServicesJmsConfiguration();
+
+        return new ArrayList<Object[]>(Arrays.asList(new Object[][]{
+                {MarshallingFormat.JAXB, jmsConfiguration},
+                {MarshallingFormat.JSON, jmsConfiguration},
+                {MarshallingFormat.XSTREAM, jmsConfiguration}
+        }));
+    }
+
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/ruleflow-group").getFile());
+    }
+
+    @Before
+    public void cleanContainers() {
+        disposeAllContainers();
+        createContainer(CONTAINER_ID, releaseId);
+    }
+
+    @Test
+    public void testExecuteSimpleRuleFlowProcessWithAsyncResponseHandler() {
+        Marshaller marshaller = MarshallerFactory.getMarshaller(new HashSet<Class<?>>(extraClasses.values()),
+                configuration.getMarshallingFormat(), client.getClassLoader());
+        ResponseCallback responseCallback = new BlockingResponseCallback(marshaller);
+        ResponseHandler asyncResponseHandler = new AsyncResponseHandler(responseCallback);
+        ruleClient.setResponseHandler(asyncResponseHandler);
+
+        List<Command<?>> commands = new ArrayList<Command<?>>();
+        commands.add(commandsFactory.newSetGlobal(LIST_NAME, new ArrayList<String>(), LIST_OUTPUT_NAME));
+        commands.add(commandsFactory.newStartProcess(PROCESS_ID));
+        commands.add(commandsFactory.newFireAllRules());
+        commands.add(commandsFactory.newGetGlobal(LIST_NAME, LIST_OUTPUT_NAME));
+        BatchExecutionCommand batchExecution = commandsFactory.newBatchExecution(commands, KIE_SESSION);
+
+        assertNoResponse(ruleClient.executeCommandsWithResults(CONTAINER_ID, batchExecution));
+        ExecutionResults result = responseCallback.get(ExecutionResults.class);
```

Comment:
Thanks a lot @mswiderski. This test was failing with JSON marshaller because of this problem.